tp1/tp1.go

```diff
@@ -24,17 +48,93 @@ type Carrito struct {
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+	// check that my list is not empty
+	if len(ids) == 0 {
+		return nil
+	}
+
+	var carritos []Carrito
+	var supermercadoByPrice = make(map[string]int)
+
+	for _, id := range ids {
+		for _, productos := range p {
+			productoId, err := strconv.Atoi(productos[ID_INDEX])
+			if err != nil {
+				fmt.Println("not a number")
+			}
+
+			if productoId == id {
```

Comment:
En go siempre se intenta manejar el "sad path" primero. Ayudar a mantener el line of sight: https://youtu.be/yeetIgNeIkc?t=256
```go
if productoId != id {
    continue
}
// codigo
```